packages/dev/core/src/Cameras/Inputs/freeCameraDeviceOrientationInput.ts

```diff
@@ -172,9 +177,9 @@ export class FreeCameraDeviceOrientationInput implements ICameraInput<FreeCamera
     };
 
     private _deviceOrientation = (evt: DeviceOrientationEvent) => {
-        this._alpha = evt.alpha !== null ? evt.alpha : 0;
-        this._beta = evt.beta !== null ? evt.beta : 0;
-        this._gamma = evt.gamma !== null ? evt.gamma : 0;
+        this._alpha = evt.alpha !== null ? Tools.SmoothAngleChange(this._alpha, evt.alpha, this._smoothFactor) : 0;
+        this._beta = evt.beta !== null ? Tools.SmoothAngleChange(this._beta, evt.beta, this._smoothFactor) : 0;
+        this._gamma = evt.gamma !== null ? Tools.SmoothAngleChange(this._gamma, evt.gamma, this._smoothFactor) : 0;

```

Comment:
SmoothAngleChange could be very expensive. It seems like there is no reason to call it if smoothFactor is 0 (which is the default). We can save those calles and only run it is the factor is > 0

Reply:
100% agree, first implementation did branch those 2 cases. I reverse to it.